macros/edr/system/system_utils/buckets_cte.sql

```diff
@@ -1,75 +1,90 @@
 {% macro complete_buckets_cte(time_bucket) %}
-    {{ adapter.dispatch('complete_buckets_cte','elementary')(time_bucket) }}
+    {%- set edr_bucket_end_expr = elementary.edr_timeadd(time_bucket.period, time_bucket.count, 'edr_bucket_start') %}
+    {%- set edr_min_bucket_start_expr = elementary.edr_date_trunc(time_bucket.period, elementary.edr_cast_as_timestamp(elementary.edr_quote(elementary.get_min_bucket_start()))) %}
+    {%- set edr_max_bucket_end_expr = elementary.edr_cast_as_timestamp(elementary.edr_quote(elementary.get_max_bucket_end())) %}
```

Comment:
Moved the expressions out of the adapters to remove duplicated code. 
Changed the min bucket start to be truncated to the start of the bucket.